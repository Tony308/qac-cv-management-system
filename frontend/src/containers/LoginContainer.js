import React, {Component} from 'react';
import LoginComponent from "../components/LoginComponent.js";

class loginContainer extends Component {

    render() {
        return(<LoginComponent/>);
    }
}

export default loginContainer;