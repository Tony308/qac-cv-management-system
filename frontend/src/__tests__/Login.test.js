import React from 'react';
import { render, cleanup} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

import LoginContainer from '../containers/LoginContainer';

afterEach(cleanup);

describe('Login container test', () => {
    test('snapshot test', () => {
        const tree = renderer.create(<LoginContainer/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('form submit without function', () => {
        const {getByRole, getByPlaceholderText} = render(<LoginContainer/>);

        const userBox = getByRole('textbox');
        const passwordBox = getByPlaceholderText('Password');
    
        userEvent.type(userBox, 'username');
        userEvent.type(passwordBox, 'password');

        expect(userBox.value).toBe('username');
        expect(passwordBox.value).toBe('password');
    })

    test('error message prop', () => {
        const { getByText } = render(<LoginContainer errMessage="error message" />);

        const message = getByText('error message');
        expect(message).toHaveTextContent('error message');

    })
})

