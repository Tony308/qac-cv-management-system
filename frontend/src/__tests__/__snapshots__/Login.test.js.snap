// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login container test snapshot test 1`] = `
<div>
  <h1>
     Login 
  </h1>
  <form
    method="POST"
    name="login-form"
  >
    <p
      style={
        Object {
          "color": "red",
          "fontSize": "13px",
        }
      }
    />
    <p>
      Username:
    </p>
    <input
      name="username"
      placeholder="Username"
      type="text"
    />
    <p>
      Password
    </p>
    <input
      name="password"
      placeholder="Password"
      type="password"
    />
    <br />
    <br />
    <input
      type="submit"
      value="Login"
    />
  </form>
</div>
`;
